lib/pure/strutils.nim

```diff
@@ -1370,17 +1371,69 @@ when not defined(js):
   proc c_sprintf(buf, frmt: cstring): cint {.header: "<stdio.h>",
                                      importc: "sprintf", varargs, noSideEffect.}
 
+# Forward declaration due to formatEng's use of ffDecimal formatting
+proc formatEng*(f: BiggestFloat,
+                precision: range[0..32] = 10,
+                decimalSep = '.',
+                trim: bool = true,
```

Comment:
I really dislike that ``trim=true`` is the default, ``0.00`` is not at all the same as ``0``. Disclaimer: I am not an engineer though.